avocado/utils/network/interfaces.py

```diff
@@ -128,6 +128,65 @@ def add_ipaddr(self, ipaddr, netmask):
         except Exception as ex:
             raise NWException("Failed to add address {}".format(ex))
 
+    @property
+    def vlans(self):
+        list_of_vlan_ids = {}
+        for i in self.host.interfaces:
+            details = i._get_interface_details()
+            if self.name == details.get("link"):
```

Comment:
Based on my previous comment, `self._get_interface_details()` is returning this specific object interface detail, so this if is not necessary.

Reply:
When we go through all of the interfaces, this if statement checks if the current interface belongs to self. i.e. 
```
$ ip -d -j link show env8.1
[{"ifindex":6,"link":"env8","ifname":"env8.1","flags":["BROADCAST","MULTICAST"]...
```
env8.1 is a vlan interface to env8. The `link` field shows that the vlan belongs to env8, from my understanding.
So when a developer calls `self.env8.vlans` it will only return the vlans of env8, not all of the vlans of the machine.